src/store/albums.js

```diff
@@ -0,0 +1,261 @@
+/**
+ * @copyright Copyright (c) 2022 Louis Chemineau <louis@chmn.me>
+ *
+ * @author Louis Chemineau <louis@chmn.me>
+ *
+ * @license AGPL-3.0-or-later
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+import { showError } from '@nextcloud/dialogs'
+import { getCurrentUser } from '@nextcloud/auth'
+
+import client from '../services/DavClient.js'
+import logger from '../services/logger.js'
+import Semaphore from '../utils/semaphoreWithPriority.js'
+
+/**
+ * @typedef {object} Album
+ * @property {string} basename - The name of the album.
+ * @property {number} lastmod - The creation date of the album.
+ * @property {string} size - The number of items in the album.
+ */
+
+const state = {
+	albums: {},
+	albumsFiles: {},
+}
+
+const mutations = {
+	/**
+	 * Add albums to the album collection.
+	 *
+	 * @param {object} state vuex state
+	 * @param {object} data destructuring object
+	 * @param {Array} data.albums list of albums
+	 */
+	addAlbums(state, { albums }) {
+		state.albums = {
+			...state.albums,
+			...albums.reduce((albums, album) => ({ ...albums, [album.basename]: album }), {}),
+		}
+	},
+
+	/**
+	 * Remove albums from the album collection.
+	 *
+	 * @param {object} state vuex state
+	 * @param {object} data destructuring object
+	 * @param {Array} data.albumNames list of albums ids
+	 */
+	removeAlbums(state, { albumNames }) {
+		albumNames.forEach(albumName => delete state.albums[albumName])
+		albumNames.forEach(albumName => delete state.albumsFiles[albumName])
+	},
+
+	/**
+	 * Add files to an album.
+	 *
+	 * @param {object} state vuex state
+	 * @param {object} data destructuring object
+	 * @param {string} data.albumName the album id
+	 * @param {string[]} data.fileIdsToAdd list of files
+	 */
+	addFilesToAlbum(state, { albumName, fileIdsToAdd }) {
+		const albumFiles = state.albumsFiles[albumName] || []
+		state.albumsFiles = {
+			...state.albumsFiles,
+			[albumName]: [
+				...albumFiles,
+				...fileIdsToAdd.filter(fileId => !albumFiles.includes(fileId)), // Filter to prevent duplicate fileId.
+			],
+		}
```

Comment:
This is redux style, I believe. With Vue(x) you should usually be able to simply use normal mutations and it will magically propagate changes. See https://v2.vuejs.org/v2/guide/reactivity.html

Reply:
Tested, and no, reactivity is only for the objects' keys, not sub keys

Reply:
For sub-keys you will need `Vue.set(object, propertyName, value)` as the link explains